src/main/java/no/unit/nva/database/RoleDb.java

```diff
@@ -0,0 +1,146 @@
+package no.unit.nva.database;
+
+import static java.util.Objects.isNull;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import java.util.Objects;
+import no.unit.nva.database.exceptions.InvalidRoleException;
+import no.unit.nva.database.intefaces.DynamoEntry;
+import no.unit.nva.database.intefaces.WithCopy;
+import no.unit.nva.database.intefaces.WithType;
+import nva.commons.utils.JacocoGenerated;
+
+@DynamoDBTable(tableName = "UsersRoles")
+public class RoleDb implements WithCopy<RoleDb.Builder>, WithType, DynamoEntry {
+
+    private static final String INVALID_PRIMARY_HASH_KEY = "PrimaryHashKey should start with \"ROLE\"";
```

Comment:
What are the access patterns / queries like for this tabe (in pseudo-code)?

Reply:
The main access pattern will be create role and update role. Probably we will not use these entries at all. Each user will have a list of roles attached to them. So all the necessary information will be included in the user. But maybe in the future we will need to add extra information in the roles